modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/actions/PreIssueAccessTokenTestCase.java

```diff
@@ -0,0 +1,591 @@
+    private static final String CUSTOM_SCOPE_3 = "test_custom_scope_3";
+    private static final String SCIM2_USERS_API = "/o/scim2/Users";
+    private static final String ACTIONS_API = "/api/server/v1/actions";
+    private static final String APPLICATION_MANAGEMENT_API = "/api/server/v1/applications";
+    private static final String API_RESOURCE_MANAGEMENT_API = "/api/server/v1/api-resources";
+    private final static Boolean REQUIRES_AUTHORIZATION = true;
+    protected OAuth2ServiceHelper oAuth2Service;
+
+    protected SCIM2RestClient scim2RestClient;
+    private String roleID;
+    private String userID;
+    private String appId;
+
+    private String consumerKeyStr;
+    private String consumerSecretStr;
+    private String accessToken;
+
+    private String[] customScopesFromToken;
+
+    private List<String> consumerKeys = new ArrayList<>();
+    private List<String> consumerSecrets = new ArrayList<>();
+    private List<String> customScopes = new ArrayList<>();
+    private List<Permission> userPermissions = new ArrayList<>();
+    @BeforeClass(alwaysRun = true)
+    public void testInit() throws Exception {
+
+        super.init(TestUserMode.TENANT_USER);
+
+        oAuth2Service = this.new OAuth2ServiceHelper();
+
+        restClient = new ActionsRestClient(serverURL, tenantInfo);
+        scim2RestClient = new SCIM2RestClient(serverURL, tenantInfo);
+        adminClient = new OauthAdminClient(backendURL, sessionCookie);
+
+        setSystemproperties();
+    }
+
+    // TODO: check what is wrong here
+//    @AfterClass(alwaysRun = true)
+//    public void atEnd() throws Exception {
+//        restClient.deleteV2Role(roleID);
+//        deleteApp(appId);
+//        scim2RestClient.deleteUser(userID);
+//        consumerKey = null;
+//        consumerSecret = null;
+//        appId = null;
+//        restClient.closeHttpClient();
+//    }
+
+    /**
+     * Provides data for testing registration pre-requisites.
+     * Each dataset consists of an external service name and its uri.
+     *
+     * @return Two-dimensional array containing pairs of external service names and its uris.
+     */
+    @DataProvider(name = "getExternalServiceInfo")
+    public Object[][] getExternalServiceInfo() {
+
+        return new Object[][]{
+                {"TestExternalService", "https://wso2is.free.beeceptor.com"}
+        };
+    }
+
+    @Test(groups = "wso2.is", description = "Registers an application with specified audience type, grant type. " +
+            "Creates a role, adds users and associates users with roles. Subscribe to necessary API resources.",
+            dataProvider = "getExternalServiceInfo")
+    private void testRegisterPreRequisites(String externalServiceName, String externalServiceURI) throws Exception {
+        appId = registerApplication(externalServiceName, externalServiceURI);
+        roleID = createRoles(appId);
+        userID = createUser();
+        assignRoleToUser(roleID, userID);
+    }
```

Comment:
Don't we have a common base class that can do above ?
If not let's move these pre stuff for the base class you implemented

Reply:
And perform before test class executes

Reply:
Fixed with the commit 6e4aab9c2f07c6fa7ac9e7d13b2ea705224a25a5